### PR TITLE
Typescript error fixes

### PR DESCRIPTION
**Pull request by GHaberis**
## What are you doing in this PR?

This PR fixes Typescript errors across a range of modules, some fixes are simple, for those that need explanation I've added some comments!

### CODE REVIEW

File: support-frontend/assets/pages/digital-subscription-checkout/components/digitalCheckoutFormGift.tsx [hunk 1]
```diff
@@ -277,7 +277,7 @@ function DigitalCheckoutFormGift(props: PropTypes): JSX.Element {
 						errorReason={props.submissionError}
 						errorHeading={submissionErrorHeading}
 					/>
-					<EndSummaryMobile orderIsAGift />
+					<EndSummaryMobile />
```
Comment (GHaberis) [inline support-frontend/assets/pages/digital-subscription-checkout/components/digitalCheckoutFormGift.tsx:280]: `orderIsAGift` no longer needs to be passed as a prop to `EndSummaryMobile`, this information is retrieved from Redux when we `mapStateToProps` for the `EndSummary`.

File: support-frontend/assets/pages/digital-subscription-checkout/components/endSummary/endSummary.tsx [hunk 2]
```diff
@@ -19,10 +19,10 @@ function Dot() {
 function EndSummary({
 	promotion,
 	priceDescription,
-	orderIsAGift = false,
+	orderIsAGift,
```
Comment (GHaberis) [inline support-frontend/assets/pages/digital-subscription-checkout/components/endSummary/endSummary.tsx:22]: This is a non-optional `prop` does not need a default value.

File: support-frontend/assets/pages/digital-subscription-checkout/components/endSummary/endSummary.tsx [hunk 3]
```diff
@@ -19,10 +19,10 @@ function Dot() {
...
 	digitalGiftBillingPeriod,
 	price,
-	isPatron = false,
+	isPatron,
```
Comment (GHaberis) [inline support-frontend/assets/pages/digital-subscription-checkout/components/endSummary/endSummary.tsx:25]: This is non-optional `prop` does not need a default value.

File: support-frontend/assets/pages/digital-subscription-checkout/components/endSummary/endSummary.tsx [hunk 4]
```diff
@@ -89,7 +89,7 @@ function NonGiftSummary(props: NonGiftSummaryTypes) {
 				<div css={styles.listMain}>{props.promotion}</div>
 				<span css={styles.subText}>{props.priceDescription}</span>
 			</li>
-			{props.isPatron && (
+			{!props.isPatron && (
```
Comment (GHaberis) [inline support-frontend/assets/pages/digital-subscription-checkout/components/endSummary/endSummary.tsx:92]: This was a bug, we were only rendering the "14 day free trial" bullet point in the end summary if `isPatron` was `true`,  however this logic was incorrect, we should only show the "14 day free trial" bullet point in the order summary if `isPatron` is `false`.

File: support-frontend/assets/pages/digital-subscription-checkout/components/endSummary/endSummary.tsx [hunk 5]
```diff
@@ -89,7 +89,7 @@ function NonGiftSummary(props: NonGiftSummaryTypes) {
 				<div css={styles.listMain}>{props.promotion}</div>
 				<span css={styles.subText}>{props.priceDescription}</span>
 			</li>
-			{props.isPatron && (
```
Comment (GHaberis) [inline support-frontend/assets/pages/digital-subscription-checkout/components/endSummary/endSummary.tsx:92]: This was a bug, we were only rendering the "14 day free trial" bullet point in the end summary if `isPatron` was `true`,  however this logic was incorrect, we should only show the "14 day free trial" bullet point in the order summary if `isPatron` is `false`.

File: support-frontend/assets/pages/digital-subscription-checkout/components/digitalCheckoutFormGift.tsx [hunk 6]
```diff
@@ -277,7 +277,7 @@ function DigitalCheckoutFormGift(props: PropTypes): JSX.Element {
 						errorReason={props.submissionError}
 						errorHeading={submissionErrorHeading}
 					/>
-					<EndSummaryMobile orderIsAGift />
```
Comment (GHaberis) [inline support-frontend/assets/pages/digital-subscription-checkout/components/digitalCheckoutFormGift.tsx:280]: `orderIsAGift` no longer needs to be passed as a prop to `EndSummaryMobile`, this information is retrieved from Redux when we `mapStateToProps` for the `EndSummary`.

File: support-frontend/assets/pages/digital-subscription-checkout/components/endSummary/endSummarySelector.ts [hunk 7]
```diff
@@ -20,12 +21,14 @@ function mapStateToProps(state: CheckoutState): EndSummaryProps {
...
 	const digitalGiftBillingPeriod =
 		billingPeriod === 'Annual' ? billingPeriod : 'Quarterly';
+
 	return {
 		priceDescription: getBillingDescription(productPrice, digitalBillingPeriod),
 		promotion: getPromotion(productPrice) ?? '',
 		orderIsAGift: orderIsAGift ?? false,
 		digitalGiftBillingPeriod,
 		price: showPrice(productPrice),
+		isPatron: userIsPatron(productPrice.promotions),
```
Comment (GHaberis) [inline support-frontend/assets/pages/digital-subscription-checkout/components/endSummary/endSummarySelector.ts:31]: `EndSummary`'s props are derived from this `mapStateToProp` function, the `EndSummary` component expects to receive a prop `isPatron`, however this function wasn't returning `isPatron`, so `EndSummary` defaulted `isPatron` to `false` always. This was a bug, I've nosw updated this `mapStateToProps` to include `isPatron`.

**Review by michaelbjacobson — APPROVED** [review]
Looks good to me! 👍🏼

**Pull request merged**